src/backend/distributed/executor/insert_select_executor.c

```diff
@@ -60,6 +62,16 @@ CoordinatorInsertSelectExecScan(CustomScanState *node)
 
 		ereport(DEBUG1, (errmsg("Collecting INSERT ... SELECT results on coordinator")));
 
+		/*
+		 * If we are dealing with partitioned table, we also need to lock its
```

Comment:
very minor `with partitioned` -> `with a partitioned`
`acquire necessary` -> `acquire the necessary`